### ISSUE TITLE
JetBrains IDE Links Not Clickable From Linter Output

### ISSUE THREAD

**Issue by nrayburn-tech** [biomejs/biome#5358]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.3"

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### Rule name

N/A

### Playground link

N/A

### Expected result

Links should be clickable in JetBrains terminals.

https://github.com/biomejs/biome/commit/d43aa7efa7ef2d91c06e103b93beac54dcedd0e4 seems to have broken this.
https://youtrack.jetbrains.com/issue/IJPL-114188/Add-environment-variable-to-terminal-detect-if-running-in-IntelliJ-IDES#focus=Comments-27-3130008.0-0 contains the env variable and value, but for easy reference it is `TERMINAL_EMULATOR=JetBrains-JediTerm`

![Image](https://github.com/user-attachments/assets/ca86e6b1-c322-42f3-82b1-712525bfe057)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
I think this is a duplicate of #4874.

It will be fixed in Biome 2.0.

**Comment by nrayburn-tech** [comment]
I’m not familiar with building biome locally, so I can’t test it however I think the env property was changed from the original PR. It should be TERMINAL_EMULATOR not TERM_PROGAM.

https://github.com/biomejs/biome/blob/5d73b1da50da81004f05c02edf4324c967028c99/crates/biome_diagnostics/src/display.rs#L125

I noticed this when comparing the implementation to what I am working on in Oxc and wanted to mention it.

**Comment by arendjr** [comment]
Thanks for the heads up! To be on the safe side, I'm checking both now: https://github.com/biomejs/biome/pull/5359
